react-app/src/components/PortfolioScreen/PortfolioScreenAPI.ts

```diff
@@ -0,0 +1,89 @@
+import { useCallback, useEffect, useMemo, useState } from "react";
+import { useCosmos } from "../../api/cosmosAPI";
+import { useQueryClient } from "../../providers/QueryClientProvider";
+import { ConnectionStatus, useWallet } from "../../providers/WalletProvider";
+import { translateAddress } from "../../utils/address";
+import { useStaking } from "../../api/stakingAPI";
+import {
+  RequestState,
+  RequestStateError,
+  RequestStateInitial,
+  RequestStateLoaded,
+  RequestStateLoading,
+} from "../../models/RequestState";
+import { Portfolio } from "./PortfolioScreenModel";
+
+type PortfolioRequestState = RequestState<Portfolio | null>;
+
+export function usePortfolioQuery(): PortfolioRequestState {
+  const [loading, setLoading] = useState(false);
+  const [data, setData] = useState<Portfolio>();
+  const [error, setError] = useState<Error>();
+
+  const wallet = useWallet();
+  const cosmosAPI = useCosmos();
+  const staking = useStaking();
+  const { desmosQuery } = useQueryClient();
+
+  const fetchPortfolio = useCallback(async () => {
+    setLoading(true);
+    if (wallet.status !== ConnectionStatus.Connected) {
+      setError(new Error("Wallet not connected!"));
+      return;
+    }
+    try {
+      const [balance, balanceStaked, balanceUnstaking, profile] =
+        await Promise.all([
+          cosmosAPI.getBalance(),
+          cosmosAPI.getBalanceStaked(),
+          staking.getAmountUnstaking(wallet.account.address),
+          desmosQuery.getProfile(
+            translateAddress(wallet.account.address, "desmos")
+          ),
+        ]);
+
+      const balanceAvailable = {
+        amount: balance.amount
+          .minus(balanceStaked.amount)
+          .minus(balanceUnstaking.amount),
+        denom: balance.denom,
+      };
+
+      setData({
+        profile,
+        balance,
+        balanceStaked,
+        balanceUnstaking,
+        balanceAvailable,
```

Comment:
@hochiw should I change these too (eg `balanceStaked` -> `stakedBalance`)?

Reply:
think so